anndata/_core/index.py

```diff
@@ -71,14 +71,22 @@ def name_idx(i):
     if is_dask(indexer):
         if is_dask(index):
             if isinstance(indexer, dd.Series):
-                indexer = slice(0, len(indexer), 1)
+                indexer = indexer.values
+            elif isinstance(indexer, da.Array):
+                pass
             else:
-                raise ValueError("Attempting to normalize dask index swith a dask indexer!  Use a dask series Indexer instead.")
-                # This works technically, but is too opaque to be useful.
-                # return daskify_call(_normalize_index, indexer, index)
+                raise ValueError("Attempting to normalize dask index with an indexer that is not a series or array: %s"
+                                 % indexer)
+            def get_indexer(*a, **kw):
+                (ix, ixr) = a
+                return ix.get_indexer(ixr)
+            pos = index.map_partitions(
+                get_indexer,
+                indexer,
+                meta=index._meta)
+            return pos
```

Comment:
This is in question.

If `indexer` and `index` were non-dask, we would do `positions = index.get_indexer(indexer)` and return `positions` as the "index" (below in the code).

Since both are dask, we need to compute both before processing.  So this is mapping over `index` partitions and passing in a fully-computed indexer, which can match the size of the index but will not exceed it.